src/holidayRecommendationService.js

```diff
@@ -0,0 +1,42 @@
+import { addDays, getNonWeekendHolidays } from './holidayService';
+import { getNumberOfWeekends, getPreviousWeekend } from './weekendService';
+
+export const getHolidayRecommendations = (year, vacationDays) => {
+  const holidayRecommendations = [];
+  const nonWeekendHolidays = getNonWeekendHolidays(year);
+
+  for (let i = 0; i < nonWeekendHolidays.length; i++) {
+    const holiday = nonWeekendHolidays[i];
+    const currentDate = new Date(holiday.date);
+
+    const startDate =
+      currentDate.getDay() >= 3 ? currentDate : getPreviousWeekend(currentDate);
+    const endDate = addDays(startDate, vacationDays);
+
+    const numberOfWeekends = getNumberOfWeekends(startDate, vacationDays);
+
+    let numberOfHolidays = 1;
+    while (
+      i < nonWeekendHolidays.length - 1 &&
+      new Date(nonWeekendHolidays[i + 1].date) <= endDate
+    ) {
+      i++;
```

Comment:
We are incrementing the index of the for loop within the loop itself. Wouldn't this effectively make the application skip certain holidays for the calculation in the main for loop? 🤔

Reply:
That was the intention of incrementing the index, I wanted to make the main loop skip the next holiday if it's in between `startDate` and `endDate`.

Reply:
But on second thought we should probably calculate all the possible cases and let the sorting take care of deciding better suggestions - will update it accordingly :)